src/cli/migrate/nodeVersion/index.ts

```diff
@@ -212,54 +172,54 @@ const writePatchedContents = async ({
   path,
   contents,
   templated,
-  test,
+  regex,
 }: {
   path: string;
   contents: string;
   templated: string;
-  test?: RegExp;
+  regex?: RegExp;
 }) =>
   await fs.promises.writeFile(
     path,
-    test ? contents.replaceAll(test, templated) : templated,
+    regex ? contents.replaceAll(regex, templated) : templated,
   );
 
-const upgrade = async (
-  { nodeVersion, nodeTypesVersion, ECMAScriptVersion }: Versions,
-  dir: string,
-) => {
+const upgrade = async (versions: Versions, dir: string) => {
   await Promise.all(
-    subPatches.map((subPatch) =>
-      runSubPatch(
-        { nodeVersion, nodeTypesVersion, ECMAScriptVersion },
-        dir,
-        subPatch,
-      ),
-    ),
+    subPatches(versions).map((subPatch) => runSubPatch(dir, subPatch)),
   );
 };
 
 export const nodeVersionMigration = async (
   {
     nodeVersion,
     ECMAScriptVersion,
-  }: { nodeVersion: number; ECMAScriptVersion: string },
+    defaultNodeTypesVersion,
+  }: {
+    nodeVersion: number;
+    ECMAScriptVersion: string;
+    defaultNodeTypesVersion: string;
+  },
   dir = process.cwd(),
 ) => {
   log.ok(`Upgrading to Node.js ${nodeVersion}`);
   try {
-    const { version: nodeTypesVersion, err } = await getNode22TypesVersion(
+    if (!(await isPatchableNodeVersion(nodeVersion))) {
+      throw new Error('Node version is not patchable');
```

Comment:
Since we seem to log the `.js` in a bunch of other spots:

```suggestion
      throw new Error('Node.js version is not patchable');
```